telegram/ext/jobqueue.py

```diff
@@ -348,6 +348,11 @@ def jobs(self):
 
         return tuple(job[1] for job in self.queue.queue if job)
 
+    def select_job(self, name):
+        """Returns a tuple of jobs with the given name that are currently in the ``JobQueue``"""
+
+        return tuple(job[1] for job in self.queue.queue if job and job[1].name==name)
```

Comment:
1. Accessing `self.queue.queue` is not thread safe. We need a different approach here.
2. code style: you're missing spaces around `==`...

Reply:
@Eldinnie I'm not sure, what do you mean about selecting none, one and multiple. So I run the following codes to test:

**Codes:**

	import time
	from telegram.ext import Updater

	TOKEN = "412548814:AAHhkKLF5rStuI1A_T5k9hb0-1qyiUQ-TY0" #fake token

	def print_job():
		pass

	def main():
		updater = Updater(TOKEN)
		j = updater.job_queue

		j.run_once(print_job, 10, name='job1')
		j.run_once(print_job, 10, name='job2')
		j.run_once(print_job, 10, name='job3')

		try:
			print('text11:', j.select_job())
		except Exception as e:
			print('test12: ',e)

		try:
			print('text21:', j.select_job(name=None))
		except Exception as e:
			print('test22: ',e)

		try:
			print('test31:', j.select_job('job1'))
		except Exception as e:
			print('test32:', e)

		try:
			print('test41', j.select_job('job1', 'job2'))
		except Exception as e:
			print('test42', e)

		try:
			print('test51:', job_queue.select_job('job4'))
		except Exception as e:
			print('test52:', e)

		print('job_list = ', j.jobs())

		time.sleep(10)

	if __name__ == '__main__':
		main()

**outputs:**

    test12:  select_job() missing 1 required positional argument: 'name'
    text21: ()
    test31: (<telegram.ext.jobqueue.Job object at 0x7fd0ae7b3828>,)
    test42 select_job() takes 2 positional arguments but 3 were given
    test51: ()
    job_list =  (<telegram.ext.jobqueue.Job object at 0x7fd0ae7b3828>, 
                 <telegram.ext.jobqueue.Job object at 0x7fd0ae7b3940>,
                 <telegram.ext.jobqueue.Job object at 0x7fd0ae7b3a58>)

@tsnoam At sleep duration I run the same script, multiple time by my machine and it seems thread safe. If this method is not thread safe so the method `jobs()` is not thread safe too, and I prefer to ignore this contribution.